tests/test_commands_windows.py

```diff
@@ -162,6 +162,21 @@ def test_flow():
     assert len(glob.glob(f"{temp_dir}\\prefect-*.ps1")) == 0
 
 
+def test_shell_run_command_throw_exception_on_nonzero_exit_code():
+    @flow
+    def test_flow():
+        return shell_run_command(
+            # ping ??? returns exit code 1
+            command="echo 'two'; echo 'lines'; ping ???", 
+            return_all=True,
+            shell="powershell")
+
+    with pytest.raises(RuntimeError):
+        result = test_flow()
+        assert result[0].rstrip() == "two"
+        assert result[1].rstrip() == "lines"
```

Comment:
When is the exception thrown? At `test_flow()`? These assertions will not run in that case which is misleading.

Reply:
It is misleading indeed.  This test should only check if a non-zero exit code raises an exception.  The integrity of the output is taken care of by other tests.  Fixed.